.github/workflows/pipeline.yml

```diff
@@ -120,10 +120,70 @@ jobs:
     - name: Perform CodeQL Analysis
       uses: github/codeql-action/analyze@v3
 
+  integration-tests:
+    name: Integration Tests
+    runs-on: 'ubuntu-latest'
+    needs: build
+    # Run on main branch or on PR
+    if: github.ref == 'refs/heads/main' || github.event_name == 'pull_request'
+    timeout-minutes: 30
+    permissions:
+      actions: read
+      contents: read
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    # Check if backend files changed (for PR only)
+    - name: Check for backend changes
+      id: backend-changes
+      if: github.event_name == 'pull_request'
+      uses: dorny/paths-filter@v2
```

Comment:
I'm not sure I want to include this action - can we implement the check ourselves in bash?